### PR TITLE
GOVSI-128: Add Google Tag Manager scripts

### PR DESCRIPTION
**Pull request by alan-gds**
## What?

Add Google Tag Manager scripts to the header in all pages, along with a hidden iframe at the top of the page body, via the base template.

## Why?

The Performance Team will be using Google Tag Manager for analytics.  I'm told the hidden iframe is part of the GTM magic.  See the JIRA ticket for more details.

### CODE REVIEW

File: src/components/common/layout/base.njk [hunk 1]
```diff
@@ -3,6 +3,20 @@
 {% from "govuk/components/phase-banner/macro.njk" import govukPhaseBanner %}
 
 {% block head %}
+    <script>
```
Comment (sanisc) [inline src/components/common/layout/base.njk:6]: this won't work as the CSP blocks any inline scripts from running for security reasons

File: src/assets/javascript/cookies.js [hunk 2]
```diff
@@ -1,8 +1,6 @@
-window.govSigin = window.govSigin ?? {};
-
-(function (w) {
-  "use strict";
+"use strict";
 
+var cookies = function (trackingId) {
```
Comment (alan-gds) [inline src/assets/javascript/cookies.js:3]: Indentation looks off here.

File: src/assets/javascript/cookies.js [hunk 3]
```diff
@@ -55,80 +38,180 @@ window.govSigin = window.govSigin ?? {};
...
+      showElement(cookiesRejected);
+    }
   }
 
-  function isOnCookiesPage() {
-    return window.location.pathname === "/cookies";
+  function saveCookieSettings(event) {
+    event.preventDefault();
+
+    const hasConsented =
```
Comment (alan-gds) [inline src/assets/javascript/cookies.js:69]: Rogue const in .js file

File: src/assets/javascript/cookies.js [hunk 4]
```diff
@@ -55,80 +38,180 @@ window.govSigin = window.govSigin ?? {};
...
+      setCookie(gtagCookie, "", { days: -1 });
+    }
+
+    window.scrollTo(0, 0);
   }
 
   function cookiesPageInit() {
-    var cookie = getCookieValue(COOKIES_PREFERENCES_SET);
-    var analyticsValue = false;
+    const analyticsConsent = hasConsentForAnalytics();
```
Comment (alan-gds) [inline src/assets/javascript/cookies.js:97]: Rogue const in .js file

**Review by alan-gds — COMMENTED** [review]
Think this is generally okay, apart from some consts being present in a .js file.  I only checked what's in the diff, and there were already a few in there, so worth doing a search to see if there are others we have missed.

I've also flagged an indentation issue but that doesn't matter much.

File: src/assets/javascript/cookies.js (hunk 2 above)
Comment (alan-gds) [reply src/assets/javascript/cookies.js:3]: My mistake; I misinterpreted the diff.

**Review by alan-gds — COMMENTED** [review]
LGTM

**Review by sanisc — APPROVED** [review]

**Pull request merged**